File: groups/group7/Lucas.js
function main(inputs) {
    return generateSequence(inputs[0], "Lucas", 2, [2n, 1n]);
}

function generateSequence(rawInput, name, minInput, initialTerms) {
    rawInput = String(rawInput ?? "").trim();
    if (!rawInput) throw new Error("Please enter a value.");
    if (rawInput.includes(".")) throw new Error("Decimals are not allowed. Please enter a whole number.");
    if (!/^\d+$/.test(rawInput)) {
        if (rawInput.startsWith("-")) throw new Error("Negative numbers are not allowed. Please enter a positive whole number.");
        throw new Error("Invalid input. Only positive whole numbers are accepted (no letters, symbols, or decimals).");
    }

    const n = parseInt(rawInput, 10);
    if (n <= minInput) throw new Error(`Input must be greater than ${minInput}. You entered: ${n}`);
    if (n > 500) throw new Error("Input exceeds the maximum allowed limit of 500 terms.");

    let seq = [...initialTerms];
    for (let i = initialTerms.length; i < n; i++) {
        let nextVal = 0n;
        for (let j = 1; j <= initialTerms.length; j++) {
            nextVal += seq[i - j];
        }
        seq.push(nextVal);
    }

    return `This program will find all the terms of the ${name} numbers.\n\tInput the number of terms: ${n}\n\tThe ${name} numbers are: ${seq.join(", ")}`;
}

File: groups/group5/Java-Recursion/Lucas/Lucas.java
package Lucas;

import java.math.BigInteger;
import java.util.ArrayList;

public class Lucas {
    public static String process(String[] inputs) throws Exception {
        int limit = Integer.parseInt(inputs[0].trim());
        if (limit <= 2)
            throw new Exception("Value must be greater than 2");
        BigInteger p2 = BigInteger.valueOf(2), p1 = BigInteger.ONE;
        ArrayList<String> res = new ArrayList<>();
        for (int i = 0; i < limit; i++) {
            if (i == 0)
                res.add(p2.toString());
            else if (i == 1)
                res.add(p1.toString());
            else {
                BigInteger curr = p2.add(p1);
                res.add(curr.toString());
                p2 = p1;
                p1 = curr;
            }
        }
        return "This program will find all the terms of the Lucas numbers.\nThe Lucas numbers are:\n"
                + String.join(", ", res);
    }
}